Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +11,246 @@ public static class BalanceStats
+            {
+                return "N/A.";
+            }
+
+            var maxBalance = 0m;
+            var name = new StringBuilder();
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+                var balance = GetLargestBalance(peopleBalances);
+                GetNamesWithHighestBalance(ref name, ref maxBalance, peopleBalances, balance);
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            return $"{FixPlural(name.ToString())} had the most money ever. {maxBalanceNoComma}.";
+        }
+
+        private static decimal GetLargestBalance(string[] peopleBalances)
+        {
+            var maxBalance = 0m;
+
+            for (var i = 1; i < peopleBalances.Length; i++)
+            {
+                var isBalanceValid = decimal.TryParse(peopleBalances[i], out var balance);
+
+                if (isBalanceValid && balance > maxBalance)
+                {
+                    maxBalance = balance;
+                }
+            }
+
+            return maxBalance;
+        }
+
+        private static void GetNamesWithHighestBalance(ref StringBuilder name, ref decimal maxBalance, string[] peopleBalances, decimal balance)
+        {
+            if (balance >= maxBalance)
+            {
+                if (balance > maxBalance)
+                {
+                    name = new StringBuilder();
+                }
+
+                if (name.Length != 0)
+                {
+                    name.Append(", ");
+                }
+
+                name.Append(peopleBalances[0]); //Builds list of names with highest balance
+                maxBalance = balance;
+            }
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return "N/A.";
+            }
+
+            var biggestLossEver = decimal.MaxValue;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+
+                if (peopleBalances.Length <= 2)
+                {
+                    continue;
+                }
+
+                var biggestLossForPerson = GetBiggestLossForPerson(peopleBalances);
+
+                if (biggestLossForPerson < biggestLossEver)
+                {
+                    biggestLossEver = biggestLossForPerson;
+                    name = peopleBalances[0];
+                }
+            }
+
+            if (name == "")
+            {
+                return "N/A.";
```

Comment:
"N/A" should be a constant, used in many places.

Reply:
I created a class constant variable, called NotAvailable, with the value "N/A", because it is used in many places.